packages/resource-deployment/scripts/function-app-create.sh

```diff
@@ -7,48 +7,89 @@
 set -eo pipefail
 
 export resourceGroupName
-export functionAppName
 export resourceName
 
 exitWithUsageInfo() {
     echo "
-Usage: $0 -r <resource group>
+Usage: $0 -r <resource group> -e <environment>
 "
     exit 1
 }
 
+addReplyUrlIfNotExist() {
+    clientId=$1
+    functionAppName=$2
+
+    # Get existing reply urls of the app registration
+    echo "Fetching existing replyUrls..."
+    replyUrls=$(az ad app show --id $clientId --query "replyUrls" -o tsv)
+    replyUrl="https://${functionAppName}.azurewebsites.net/.auth/login/aad/callback"
+
+    for url in $replyUrls; do
+        if [[ $url == $replyUrl ]]; then
+            echo "replyUrl ${replyUrl} already exsits."
+            return
+        fi
+    done
+
+    echo "Adding replyUrl $replyUrl to app registration..."
+    az ad app update --id $clientId --add replyUrls $replyUrl
+    echo "  Successfully added replyUrl."
+}
+
+createAppRegistration() {
+    resourceGroupName=$1
+    environment=$2
+
+    appRegistrationName="allyappregistration-$resourceGroupName-$environment"
+    echo "Creating a new AppRegistration with display name $appRegistrationName..."
+    clientId=$(az ad app create --display-name "$appRegistrationName" --query "appId" -o tsv)
+    echo "  Successfully created '$appRegistrationName' App Registration with Client ID '$clientId'"
+}
+
 # Set default ARM Function App template files
 templateFilePath="${0%/*}/../templates/function-app-template.json"
 
 # Read script arguments
-while getopts "r:" option; do
+while getopts "r:e:" option; do
     case $option in
     r) resourceGroupName=${OPTARG} ;;
+    e) environment=${OPTARG} ;;
     *) exitWithUsageInfo ;;
     esac
 done
 
-if [ -z "$resourceGroupName" ]; then
+if [ -z $resourceGroupName ] || [ -z $environment ]; then
     exitWithUsageInfo
 fi
 
-# Start deployment
+# Will return the function name if we deployed before the function app template on this resource group
+echo "Checking if the function app and AppRegistration already exists..."
+functionAppName=$(az group deployment show -g "$resourceGroupName" -n "function-app-template" --query "properties.parameters.name.value" -o tsv)
+clientId=$(az webapp auth show -n "$functionAppName" -g "$resourceGroupName" --query "clientId" -o tsv) || true
```

Comment:
do we need to check first if the function name is valid and it still exists 
"[ ! -z "$functionAppName" ] && (az functionapp show -n "$functionAppName" -g "$resourceGroupName" 1>/dev/null)"
if it dowesn't, did we try if this is not gonna throw exception?

Reply:
if the function app is deleted, clientId will be empty.